pyirf/interpolation.py

```diff
@@ -54,45 +55,118 @@ def interpolate_effective_area_per_energy_and_fov(
+        Interpolated histograms
 
-    # interpolation
-    matrix_interp = griddata(grid_points, energy_dispersions, target_point, method=method)
-
-    # now we need to renormalize along the migration axis
-    norm = np.sum(matrix_interp, axis=1, keepdims=True)
-    # By using out and where, it is ensured that columns with norm = 0 will have 0 values without raising an invalid value warning
-    mig_norm = np.divide(matrix_interp, norm, out=np.zeros_like(matrix_interp), where=norm != 0)
-    return mig_norm
+    References
+    ----------
+    .. [1] B. E. Hollister and A. T. Pang (2013). Interpolation of Non-Gaussian Probability Distributions
+           for Ensemble Visualization
+           https://engineering.ucsc.edu/sites/default/files/technical-reports/UCSC-SOE-13-13.pdf
+    """
+    # To have the needed axis always at the last index, as the number of indices is
+    # not safely propageted through the interpolation but the last element remains the last element
+    hists = np.swapaxes(hists, axis, -1)
+
+    # Compute CDFs
+    cdfs = np.cumsum(hists, axis=-1)
+
+    cdfs = cdfs / np.expand_dims(np.max(cdfs, axis=-1), axis=-1)
+
+    quantiles = np.linspace(0, 1, int(np.round(1 / quantile_resolution, decimals=0)))
+
+    bin_mids = bin_center(edges)
+
+    # create ppf values from cdf samples via interpolation of the cdfs, determine quantile steps of [1]
+    ppfs_resampled = np.apply_along_axis(
+        lambda cdf: interp1d(
+            cdf, bin_mids, bounds_error=False, fill_value="extrapolate"
+        )(quantiles),
+        -1,
+        cdfs,
+    )
+
+    # nD interpolation of ppf values, interpolate quantiles step of [1]
+    ppf_interpolant = griddata(m, ppfs_resampled, mprime)
+
+    # recalculate pdf values, evaluate interpolant PDF values step of [1]
+    pdf_interpolant = np.diff(quantiles) / np.diff(ppf_interpolant, axis=-1)
+
+    # Unconventional solution to make this usable with np.apply_along_axis for readability
+    # The ppf bin-mids are computed since the pdf-values are derived through derivation
+    # from the ppf-values
+    xyconcat = np.concatenate(
+        (
+            ppf_interpolant[..., :-1] + np.diff(ppf_interpolant) / 2,
+            np.nan_to_num(pdf_interpolant),
+        ),
+        axis=-1,
+    )
+
+    # Interpolate pdf samples and evaluate at bin edges, weight with the bin_width to estimate
+    # correct bin height via the midpoint rule formulation of the trapezoidal rule
+    result = np.apply_along_axis(
+        lambda xy: np.diff(edges)
+        * np.nan_to_num(
+            interp1d(
+                xy[: int(len(xy) / 2)],
+                xy[int(len(xy) / 2) :],
+                bounds_error=False,
+                fill_value=(0, 0),
+            )(bin_mids)
+        ),
+        -1,
+        xyconcat,
+    )
+
+    # Renormalize histogram to a sum of 1
+    norm = np.sum(result, axis=-1)
+    norm = np.repeat(np.expand_dims(norm, -1), result.shape[-1], axis=-1)
```

Comment:
Why is repeating / expanding dims necessary here? Should be broadcasting in the division below be enough?

Reply:
```
norm = np.sum(result, axis=1)
with np.errstate(invalid="ignore"):
    result /= norm[..., np.newaxis]
result = np.nan_to_num(result)
```